source/Octopus.Tentacle.Tests/Integration/ScriptServiceV2Fixture.cs

```diff
@@ -0,0 +1,465 @@
+                .WithScriptBody(script)
+                .Build();
+
+            var durationUntilScriptStartedRunning = Stopwatch.StartNew();
+            var response = service.StartScript(startScriptCommand);
+
+            while (response.State != ProcessState.Complete)
+            {
+                response = service.GetStatus(new ScriptStatusRequestV2(startScriptCommand.ScriptTicket, response.NextLogSequence));
+
+                if (response.State == ProcessState.Running && durationUntilScriptStartedRunning.IsRunning)
+                {
+                    durationUntilScriptStartedRunning.Stop();
+                }
+
+                if (response.State != ProcessState.Complete)
+                {
+                    await Task.Delay(TimeSpan.FromMilliseconds(100));
+                }
+            }
+
+            service.CompleteScript(new CompleteScriptCommandV2(startScriptCommand.ScriptTicket));
+
+            durationUntilScriptStartedRunning.Elapsed.Should().BeLessOrEqualTo(TimeSpan.FromSeconds(10));
+        }
+
+        [Test]
+        public async Task ShouldExecuteALongRunningScriptSuccessfully()
+        {
+            var bashScript = "sleep 10";
+            var windowsScript = "Start-Sleep -Seconds 10";
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithScriptBodyForCurrentOs(windowsScript, bashScript)
+                .Build();
+
+            var startScriptResponse = service.StartScript(startScriptCommand);
+            var (_, finalResponse) = await RunUntilScriptCompletes(startScriptCommand, startScriptResponse);
+
+            finalResponse.State.Should().Be(ProcessState.Complete);
+            finalResponse.ExitCode.Should().Be(0);
+        }
+
+        [Test]
+        public void StartScriptShouldWaitForAShortScriptToFinish()
+        {
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithScriptBody("echo \"finished\"")
+                .WithDurationStartScriptCanWaitForScriptToFinish(TimeSpan.FromSeconds(5))
+                .Build();
+
+            var startScriptResponse = service.StartScript(startScriptCommand);
+
+            startScriptResponse.State.Should().Be(ProcessState.Complete);
+            startScriptResponse.ExitCode.Should().Be(0);
+            startScriptResponse.Logs.Count.Should().BeGreaterThan(1);
+
+            service.CompleteScript(new CompleteScriptCommandV2(startScriptCommand.ScriptTicket));
+        }
+
+        [Test]
+        public async Task StartScriptShouldNotWaitForALongScriptToFinish()
+        {
+            var bashScript = "sleep 10";
+            var windowsScript = "Start-Sleep -Seconds 10";
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
+                .WithScriptBodyForCurrentOs(windowsScript, bashScript)
+                .WithDurationStartScriptCanWaitForScriptToFinish(TimeSpan.FromSeconds(5))
```

Comment:
Consider dropping the wait time to 1s, tests that work based on timings can be flaky anything we can do to mitigate that is worthwhile.